automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/BatchNodepoolCreationSpec.scala

```diff
@@ -0,0 +1,144 @@
+package org.broadinstitute.dsde.workbench.leonardo
+
+import java.nio.file.Paths
+
+import cats.effect.IO
+import cats.implicits._
+import com.google.container.v1.{Cluster, NodePool}
+import org.broadinstitute.dsde.workbench.DoneCheckable
+import org.broadinstitute.dsde.workbench.auth.AuthToken
+import org.broadinstitute.dsde.workbench.google2.GKEModels.{KubernetesClusterId, KubernetesClusterName}
+import org.broadinstitute.dsde.workbench.google2.{streamFUntilDone, DiskName, GKEService}
+import org.broadinstitute.dsde.workbench.leonardo.LeonardoApiClient._
+import org.broadinstitute.dsde.workbench.leonardo.http.{GetAppResponse, ListAppResponse, PersistentDiskRequest}
+import org.http4s.headers.Authorization
+import org.http4s.{AuthScheme, Credentials}
+import org.scalatest.{DoNotDiscover, ParallelTestExecution}
+
+import scala.collection.JavaConverters._
+import scala.concurrent.duration._
+
+@DoNotDiscover
+class BatchNodepoolCreationSpec
+    extends GPAllocFixtureSpec
+    with LeonardoTestUtils
+    with GPAllocUtils
+    with ParallelTestExecution {
+
+  implicit val ronToken: AuthToken = ronAuthToken
+  implicit val auth: Authorization = Authorization(Credentials.Token(AuthScheme.Bearer, ronCreds.makeAuthToken().value))
+
+  val gkeServiceResource = GKEService.resource(Paths.get(LeonardoConfig.GCS.pathToQAJson), blocker, semaphore)
+
+  //the cluster and nodepools should be running
+  val clusterDoneCheckable: DoneCheckable[Option[Cluster]] =
+    x =>
+      x.map(_.getStatus()) == Some(Cluster.Status.RUNNING) &&
+        x.map(_.getNodePoolsList().asScala.toList.map(_.getStatus()).distinct) == Some(List(NodePool.Status.RUNNING))
+
+  "batch nodepool creation should work" - {
+    withNewProject { googleProject =>
```

Comment:
for now, all tests use this lightweight `withNewProject` since there are project restrictions leo imposes (namely one cluster). Once other currently WIP galaxy tests are done, we plan to make a new suite with a beforeAndAfterEach for this purpose.